### PR TITLE
feat(starfish): Use span light when txn not included

### PR DESCRIPTION
**Pull request by wmak**
- The span querybuilder will use the span self time light metric when transaction is not a part of the conditions or a selected column
- This new metric doesn't have the transaction tag which means that in these situations the queries will run a lot faster
- This means that the starfish homepage will only have data from 12pm utc, june 29th

### CODE REVIEW

File: src/sentry/search/events/builder/spans_metrics.py [hunk 1]
```diff
@@ -22,6 +24,24 @@ def get_field_type(self, field: str) -> Optional[str]:
 
         return None
 
+    def resolve_select(
+        self, selected_columns: Optional[List[str]], equations: Optional[List[str]]
+    ) -> List[SelectType]:
+        if selected_columns and "transaction" in selected_columns:
+            self.has_transaction = True
```
Comment (shruthilayaj) [inline src/sentry/search/events/builder/spans_metrics.py:31]: We should also add this to `resolve_filter` as well 👀

**Review by shruthilayaj — APPROVED** [review]
sweet!

File: src/sentry/search/events/builder/spans_metrics.py (hunk 1 above)
Comment (wmak) [reply src/sentry/search/events/builder/spans_metrics.py:31]: So I'm actually sorta cheating it here cause `resolve_conditions` happens before `resolve_select` it'll call `resolve_metric_index` which is what L38 is doing.

Unfortunately that won't work for the `selected_columns` cause nothing in there depends on order (and i didn't want to make it depend on order) so I have to check the raw column name here instead

**Review by k-fish — DISMISSED** [review]

**Pull request merged**